src/librustdoc/json/conversions.rs

```diff
@@ -652,27 +685,31 @@ impl FromClean<clean::Impl> for Impl {
                 .into_iter()
                 .map(|x| x.to_string())
                 .collect(),
-            trait_: trait_.as_ref().map(|path| path.into_json(renderer)),
+            trait_: trait_.into_json(renderer),
             for_: for_.into_json(renderer),
             items: renderer.ids(&items),
             is_negative,
             is_synthetic,
-            blanket_impl: blanket_impl.map(|x| x.as_ref().into_json(renderer)),
+            blanket_impl: blanket_impl.map(|x| x.into_json(renderer)),
         }
     }
 }
 
-pub(crate) fn from_function(
-    clean::Function { decl, generics }: &clean::Function,
-    has_body: bool,
-    header: rustc_hir::FnHeader,
-    renderer: &JsonRenderer<'_>,
-) -> Function {
-    Function {
-        sig: decl.into_json(renderer),
-        generics: generics.into_json(renderer),
-        header: from_fn_header(&header),
-        has_body,
+impl FromClean<(&Box<clean::Function>, bool, rustc_hir::FnHeader)> for Function {
```

Comment:
I don't like implementing `FromClean<(...)>`, and using a tuple for passing multiple arguments when the `clean` types don't line up with the `rustdoc_json_types` types. It's very stylistic, but I think it's just obscure and obtuse.

I don't think we need to centralize everything into happening via `FromClean`, and it's fine for there to be a few free-floating functions like this. (Expecailly as this doesn't need to get called via the "combinator" impls on Option/Box/Vec).